File: Stack/Stack.c
#include "Stack.h"
#include <stdlib.h>
#include <stdio.h>



int* Stackarray = NULL;
int top_of_stack = -1;

void create()
{
    destroy();
    Stackarray = (int*)malloc(MAX_STACK_SIZE * sizeof(int));
    top_of_stack = -1;
    return;
}

void destroy(){
    if(NULL != Stackarray)
    {
        free(Stackarray);
        Stackarray = NULL;
    }
    return;
}

bool push(int val){
    if(False == Check_if_full())
    {
        top_of_stack++;
        Stackarray[top_of_stack] = val;
        return True;
    }
    return False;
}

bool pop(int* Ptr){
    if(False == Check_if_empty())
    {
        *Ptr = Stackarray[top_of_stack];
        top_of_stack--;
        return True;
    }
    return False;
}

bool peek(int* Ptr){
    if(False == Check_if_empty())
    {
        *Ptr = Stackarray[top_of_stack];
        //top_of_stack--;
        return True;
    }
    return False;
}

int size()
{
    return top_of_stack + 1;
}

bool Check_if_empty()
{
    if(-1 == top_of_stack)
        return True;

    return False;
}

bool Check_if_full()
{
    if(MAX_STACK_SIZE == top_of_stack + 1)
        return True;

    return False;
}

void Out_Stack()
{
    if(True == Check_if_empty())
        return;
    printf("***************\n");
    for(int i = top_of_stack; -1 < i; i--)
    {
        printf("at position %d ->  %d\n", i, Stackarray[i]);
    }
    printf("***************\n");
}


// Main function is commented out

int main()
{
    create();

    push(100);
    push(20);
    push(40);
    push(50);
    push(90);
    push(95);
    Out_Stack();
    int Num = 0;

    if(pop(&Num) == True)
        printf("%d\n",Num);
    
    if(pop(&Num) == True)
        printf("%d\n",Num);
    
    if(peek(&Num) == True)
        printf("%d\n",Num);
    
    if(peek(&Num) == True)
        printf("%d\n",Num);
    
    if(pop(&Num) == True)
        printf("%d\n",Num);
    
    Out_Stack();
    destroy();
    return 0;
}

File: Stack/Stack.py

class Stack:
    
    # Stack using python list is implemented
    # a list and top of stack is initialised in constructor
    def __init__(self) -> None:
        self.__Arr = list()
        self.__Top_of_Stack = -1

    # ifempty method is defined to check if stack is empty.
    def __ifempty(self):
        if(self.__Arr == []):
            return True
        return False


    # push method is defined to push element in stack.
    def push(self, Val):
        self.__Top_of_Stack = self.__Top_of_Stack + 1
        self.__Arr.insert(self.__Top_of_Stack,Val)
    

    # peek method just outputs current top of stack.
    def peek(self):
        if(not self.__ifempty()):
            return self.__Arr[self.__Top_of_Stack]
        else:
            print("Stack is empty")

    # pop method removes top of stack from the stack and next element is made top of stack
    def pop(self):
        if(not self.__ifempty()):
            Val = self.__Arr.pop(self.__Top_of_Stack)
            self.__Top_of_Stack = self.__Top_of_Stack - 1
            return Val
        else:
            print("Stack is empty")

    # len method outputs length of stack.
    def len(self):
        return len(self.__Arr)

    # out_stack method outputs th current stack.
    def out_stack(self):
        for i in range(self.__Top_of_Stack , -1 , -1):
            print("At position ",i," element is : ",self.__Arr[i])

    # this is destructor 
    #def __del__(self):
        #print("Stack is destroyed")
    

# using stack
"""
stack = Stack()
stack.push(100)
stack.push(90)
stack.push(80)
stack.push(70)
stack.push(60)

stack.out_stack()

print(stack.peek())
print(stack.pop())
print(stack.peek())

stack.out_stack()"""
